packages/alfa-dom/src/node/attribute.ts

```diff
@@ -93,6 +96,29 @@ export class Attribute extends Node {
     return path;
   }
 
+  /**
+   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#space-separated-tokens
+   */
+  public tokens(separator: string | RegExp = /\s+/): Array<string> {
+    return this._value.trim().split(separator).filter(not(isEmpty));
+  }
+
+  /**
+   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
+   */
+  public enumerate(
+    ...keywords: Array<string>
+  ): Result<string, Attribute.EnumeratedAttributeError> {
+    if (this._value === "")
+      return Err.of(Attribute.EnumeratedAttributeError.Missing);
+
+    const value = this._value.toLowerCase();
+
+    return keywords.length === 0 || keywords.includes(value)
+      ? Ok.of(value)
+      : Err.of(Attribute.EnumeratedAttributeError.Invalid);
+  }
```

Comment:
```suggestion
  /**
   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
   */
  public enumerate(): Option<string>;

  /**
   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
   */
  public enumerate(valid: string, ...rest: Array<string>): Option<string>;
  
  public enumerate(
    ...valid: Array<string>
  ): Option<string> {
    const value = this._value.toLowerCase();

    return valid.length === 0 || valid.includes(value)
      ? Some.of(value)
      : None;
  }
```

A return value of `None` will always mean "invalid"; it's only when the attribute _isn't_ specified that the "missing" state is relevant. If it's specified with an empty value then it's still considered specified. As it turns out, the empty string can actually be a valid keyword, as noted in the spec, so we shouldn't return `None` when the attribute value is empty.

Reply:
I had somehow missed the fact that empty string can be a keyword 🤦 
This indeed makes things much easier with each error coming from a single and separate source…